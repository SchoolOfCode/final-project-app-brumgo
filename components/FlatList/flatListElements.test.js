import React from "react";
import * as renderer from "react-test-renderer";
import Component from "./index";

describe("Header Section", () => {
  it("should match to snapshot", () => {
    const component = renderer.create(<Component />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
