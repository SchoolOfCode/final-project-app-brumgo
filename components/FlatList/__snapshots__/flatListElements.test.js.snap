// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Section should match to snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#28559A",
      "borderRadius": 10,
      "flex": 1,
      "width": "100%",
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "category": "Building",
          "poiLocation": Object {
            "lat": 52.477074,
            "lng": -1.893224,
          },
          "poiName": "St Martin in the Bullring",
          "poiTrivia": "The present Victorian church was built on the site of a 13th-century predecessor, which was documented in 1263. The church was enlarged in medieval times and the resulting structure consisted of a lofty nave and chancel, north and south aisles and a northwest tower with spire.",
        },
        Object {
          "category": "Market",
          "poiLocation": Object {
            "lat": 52.476499,
            "lng": -1.892838,
          },
          "poiName": "Bull Ring Markets",
          "poiTrivia": "One of the oldest continuously trading markets in the world, The market legally began in 1154 when Peter de Bermingham, a local landowner, obtained a Charter of Marketing Rights from King Henry II.",
        },
        Object {
          "category": "Event",
          "poiLocation": Object {
            "lat": 52.47488727,
            "lng": -1.88596523,
          },
          "poiName": "Battle of Birmingham",
          "poiTrivia": "During the Civil War the Battle of Birmingham was fought up into the town through Digbeth High Street on Easter Monday 3 April 1643",
        },
        Object {
          "category": "Building",
          "poiLocation": Object {
            "lat": 52.4752029,
            "lng": -1.884587,
          },
          "poiName": "The Custard Factory",
          "poiTrivia": "The Custard Factory complex is set in fifteen acres (60,000 m²) of factory buildings, originally constructed by Sir Alfred Frederick Bird (1849–1922), the son of Alfred Bird (1811–1878), the inventor of egg-free custard.",
        },
        Object {
          "category": "Building",
          "poiLocation": Object {
            "lat": 52.47481175,
            "lng": -1.88355471,
          },
          "poiName": "The Old Crown",
          "poiTrivia": "The Old Crown pub is the oldest secular building in Birmingham, England. It is Grade II listed, and claims to date back to 1368, retaining its 'black and white' timber frame, although almost all of the present building dates from the early 16th century.",
        },
        Object {
          "category": "Building",
          "poiLocation": Object {
            "lat": 52.478311,
            "lng": -1.887021,
          },
          "poiName": "Old Typhoo Tea Factory",
          "poiTrivia": "One of the nation’s most loved teabag brands formed in Birmingham in the late 19th century. The building survived bombing during the Second World War, but was eventually closed in 1978.",
        },
        Object {
          "category": "Culture",
          "poiLocation": Object {
            "lat": 52.474774,
            "lng": -1.885914,
          },
          "poiName": "Irish Centre",
          "poiTrivia": "Digbeth hosts the largest St. Patrick’s Day parade in the country. Appoximately 100,000 people line the streets of Digbeth one Sunday each March for the ever-popular Paddy’s Day Parade. Birmingham has always had a strong connection with Ireland, which is proven by the popularity of the parade!",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#bcbbc1",
                  "flexDirection": "row",
                  "padding": 14,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "fontSize": 17,
                    }
                  }
                  testID="listItemTitle"
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                >
                  St Martin in the Bullring
                </Text>
              </View>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 15,
                "letterSpacing": 1,
                "padding": 15,
              }
            }
          >
            The present Victorian church was built on the site of a 13th-century predecessor, which was documented in 1263. The church was enlarged in medieval times and the resulting structure consisted of a lofty nave and chancel, north and south aisles and a northwest tower with spire.
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#bcbbc1",
                  "flexDirection": "row",
                  "padding": 14,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "fontSize": 17,
                    }
                  }
                  testID="listItemTitle"
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                >
                  Bull Ring Markets
                </Text>
              </View>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 15,
                "letterSpacing": 1,
                "padding": 15,
              }
            }
          >
            One of the oldest continuously trading markets in the world, The market legally began in 1154 when Peter de Bermingham, a local landowner, obtained a Charter of Marketing Rights from King Henry II.
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#bcbbc1",
                  "flexDirection": "row",
                  "padding": 14,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "fontSize": 17,
                    }
                  }
                  testID="listItemTitle"
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                >
                  Battle of Birmingham
                </Text>
              </View>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 15,
                "letterSpacing": 1,
                "padding": 15,
              }
            }
          >
            During the Civil War the Battle of Birmingham was fought up into the town through Digbeth High Street on Easter Monday 3 April 1643
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#bcbbc1",
                  "flexDirection": "row",
                  "padding": 14,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "fontSize": 17,
                    }
                  }
                  testID="listItemTitle"
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                >
                  The Custard Factory
                </Text>
              </View>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 15,
                "letterSpacing": 1,
                "padding": 15,
              }
            }
          >
            The Custard Factory complex is set in fifteen acres (60,000 m²) of factory buildings, originally constructed by Sir Alfred Frederick Bird (1849–1922), the son of Alfred Bird (1811–1878), the inventor of egg-free custard.
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#bcbbc1",
                  "flexDirection": "row",
                  "padding": 14,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "fontSize": 17,
                    }
                  }
                  testID="listItemTitle"
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                >
                  The Old Crown
                </Text>
              </View>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 15,
                "letterSpacing": 1,
                "padding": 15,
              }
            }
          >
            The Old Crown pub is the oldest secular building in Birmingham, England. It is Grade II listed, and claims to date back to 1368, retaining its 'black and white' timber frame, although almost all of the present building dates from the early 16th century.
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#bcbbc1",
                  "flexDirection": "row",
                  "padding": 14,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "fontSize": 17,
                    }
                  }
                  testID="listItemTitle"
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                >
                  Old Typhoo Tea Factory
                </Text>
              </View>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 15,
                "letterSpacing": 1,
                "padding": 15,
              }
            }
          >
            One of the nation’s most loved teabag brands formed in Birmingham in the late 19th century. The building survived bombing during the Second World War, but was eventually closed in 1978.
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#bcbbc1",
                  "flexDirection": "row",
                  "padding": 14,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "fontSize": 17,
                    }
                  }
                  testID="listItemTitle"
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                >
                  Irish Centre
                </Text>
              </View>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 15,
                "letterSpacing": 1,
                "padding": 15,
              }
            }
          >
            Digbeth hosts the largest St. Patrick’s Day parade in the country. Appoximately 100,000 people line the streets of Digbeth one Sunday each March for the ever-popular Paddy’s Day Parade. Birmingham has always had a strong connection with Ireland, which is proven by the popularity of the parade!
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
